integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/AbstractAutomaticIndexingBridgeIT.java

```diff
@@ -21,71 +17,37 @@
+import org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings;
+import org.hibernate.search.mapper.orm.mapping.HibernateOrmMappingDefinitionContainerContext;
+import org.hibernate.search.mapper.orm.mapping.HibernateOrmSearchMappingConfigurer;
 import org.hibernate.search.mapper.pojo.bridge.PropertyBridge;
 import org.hibernate.search.mapper.pojo.bridge.TypeBridge;
-import org.hibernate.search.mapper.pojo.bridge.binding.PropertyBridgeBindingContext;
-import org.hibernate.search.mapper.pojo.bridge.binding.TypeBridgeBindingContext;
-import org.hibernate.search.mapper.pojo.bridge.declaration.PropertyBridgeMapping;
-import org.hibernate.search.mapper.pojo.bridge.declaration.PropertyBridgeRef;
-import org.hibernate.search.mapper.pojo.bridge.declaration.TypeBridgeMapping;
-import org.hibernate.search.mapper.pojo.bridge.declaration.TypeBridgeRef;
-import org.hibernate.search.mapper.pojo.bridge.runtime.PropertyBridgeWriteContext;
-import org.hibernate.search.mapper.pojo.bridge.runtime.TypeBridgeWriteContext;
 import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
-import org.hibernate.search.mapper.pojo.model.PojoElement;
-import org.hibernate.search.mapper.pojo.model.PojoModelElementAccessor;
-import org.hibernate.search.mapper.pojo.model.PojoModelType;
 import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
 import org.hibernate.search.util.impl.integrationtest.orm.OrmSetupHelper;
 import org.hibernate.search.util.impl.integrationtest.orm.OrmUtils;
 import org.hibernate.search.util.impl.test.annotation.TestForIssue;
 
-import org.junit.Before;
+import org.junit.Assume;
 import org.junit.Rule;
 import org.junit.Test;
 
 /**
- * Test automatic indexing based on Hibernate ORM entity events when
+ * An abstract base for tests dealing with automatic indexing based on Hibernate ORM entity events when
  * {@link TypeBridge}s or {@link PropertyBridge}s are involved.
  */
-public class AutomaticIndexingBridgeIT {
+public abstract class AbstractAutomaticIndexingBridgeIT {
 
 	@Rule
 	public BackendMock backendMock = new BackendMock( "stubBackend" );
 
 	@Rule
 	public OrmSetupHelper ormSetupHelper = new OrmSetupHelper();
 
-	private SessionFactory sessionFactory;
-
-	@Before
-	public void setup() {
-		backendMock.expectSchema( IndexedEntity.INDEX, b -> b
-				.objectField( "typeBridge", b2 -> b2
-						.field( "directField", String.class )
-						.objectField( "child", b3 -> b3
-								.field( "includedInTypeBridge", String.class )
-						)
-				)
-				.objectField( "propertyBridge", b2 -> b2
-						.field( "includedInPropertyBridge", String.class )
-				)
-		);
-
-		sessionFactory = ormSetupHelper.withBackendMock( backendMock )
-				.setup(
-						IndexedEntity.class,
-						ContainedEntity.class
-				);
-		backendMock.verifyExpectationsMet();
-	}
-
 	@Test
 	public void directPersistUpdateDelete() {
+		SessionFactory sessionFactory = setupWithTypeBridge();
```

Comment:
So that I can more easily have multiple variants of this tests: one using accessors, one using explicit dependency declaration, one using explicit reindexing declaration.

Currently I need 4 classes, if I split this in two test classes I would need 8 classes.